src/main/java/io/spokestack/spokestack/SpeechPipeline.java

```diff
@@ -340,6 +342,60 @@ public Builder setProperty(String key, Object value) {
             return this;
         }
 
+        /**
+         * applies configuration from a {@link PipelineProfile} to the current
+         * builder, returning the modified builder. subsequent calls to {@code
+         * useProfile} or {@code setProperty} can override configuration set by
+         * a profile.
+         *
+         * @param profileClass class name of the profile to apply.
+         * @return an updated builder
+         * @throws IllegalArgumentException if the specified profile does not
+         *                                  exist
+         */
+        public Builder useProfile(String profileClass)
+              throws IllegalArgumentException {
+            return this.useProfile(profileClass, null);
+        }
+
+        /**
+         * applies configuration from a {@link PipelineProfile} to the current
+         * builder, returning the modified builder. subsequent calls to {@code
+         * useProfile} or {@code setProperty} can override configuration set by
+         * a profile.
+         *
+         * @param profileClass  class name of the profile to apply.
+         * @param profileConfig additional configuration properties associated
+         *                      with the profile.
+         * @return an updated builder
+         * @throws IllegalArgumentException if the specified profile does not
+         *                                  exist
+         */
+        public Builder useProfile(String profileClass,
```

Comment:
I think this override should be removed
there is already a mechanism for applying configuration to the pipeline, using the builder's setProperty method
and if a profile needs to set configuration on the builder, it can just call setProperty on the builder